### PR TITLE
OSDOCS-349 Update glossary reference from router to Ingress

### PR DESCRIPTION
**Pull request by bmcelvee**
https://jira.coreos.com/browse/OSDOCS-349

### CODE REVIEW

**Comment by bmcelvee** [comment]
@Miciah would you mind taking a look please? Just want to make sure the definition is still correct. Thanks!

File: contributing_to_docs/term_glossary.adoc [hunk 1]
```diff
@@ -386,17 +386,17 @@ Pod are running at all times. Used by DeploymentConfigs.
 Do not use: replication controller(s)
 
 ''''
-=== Route
+=== Ingress
 
-Usage: Route(s)
+Usage: Ingress, Ingress Controller(s)
 
 OpenShift-specific API object that allows developers to expose Services through
```
Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: `Ingress` is not OpenShift-specific; it comes from Kubernetes.

File: contributing_to_docs/term_glossary.adoc [hunk 2]
```diff
@@ -386,17 +386,17 @@ Pod are running at all times. Used by DeploymentConfigs.
...
 an HTTP(S) aware load balancing and proxy layer via a public DNS entry. The
-Route may further specify TLS options and a certificate, or specify a public
-CNAME that the OpenShift router should also accept for HTTP and HTTPS traffic.
-An administrator typically configures their router to be visible outside the
-cluster firewall, and may also add additional security, caching, or traffic
-controls on the Service content.
+Ingress Controller may further specify TLS options and a certificate, or specify
```
Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: ```Ingress Controller```→``` `Ingress` resource ``` on this line.  (Technically, the `Ingress` resource may specify _the name of a secret that contains_ the certificate, whereas the `Route` resource specifies the certificate itself, but it is probably safe to gloss over that detail.)

File: contributing_to_docs/term_glossary.adoc [hunk 3]
```diff
@@ -386,17 +386,17 @@ Pod are running at all times. Used by DeploymentConfigs.
...
+a public CNAME that the OpenShift Ingress Controller should also accept for HTTP
+and HTTPS traffic. An administrator typically configures their router to be
```
Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: "router"→"Ingress Controller".

File: contributing_to_docs/term_glossary.adoc [hunk 4]
```diff
@@ -386,17 +386,17 @@ Pod are running at all times. Used by DeploymentConfigs.
 Do not use: replication controller(s)
 
 ''''
-=== Route
+=== Ingress
```
Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: The entry for "Ingress" needs to be moved to maintain alphabetization.

**Review by Miciah — COMMENTED** [review]
The `Route` resource is not going away, so we should document both the `Route` resource and the `Ingress` resource.  However, we are moving away from the "router" terminology for the component, which we now instead refer to as the "ingress controller".  It may be helpful to have an entry for "router" too though, if only to refer the reader to the new name.  So I suggest the following entries:

* `Ingress` resource (the new entry, with some changes I suggest in comments).
* `Route` resource (the old entry).
* The ingress controller.
* The router ("Old name for the ingress controller, q.v.").

**Comment by bmcelvee** [comment]
Thanks so much, @Miciah! I separated the Ingress and Route resources. Would it be okay to add an entry for Ingress Controller where we state that it's replacing router?

**Comment by Miciah** [comment]
> Would it be okay to add an entry for Ingress Controller where we state that it's replacing router?

That would make sense to me.  Also, we should distinguish between an Ingress Controller and an `Ingress` resource: a developer creates an `Ingress` resource to expose a `Service`, and the Ingress Controller is what actually forwards traffic to the `Service` (or to be more technically accurate, it forwards traffic to the endpoints of the `Service`).

**Comment by bmcelvee** [comment]
@openshift/team-documentation PTAL, thanks! :)

File: contributing_to_docs/term_glossary.adoc [hunk 5]
```diff
@@ -213,6 +213,25 @@ The file that Ignition uses to configure Red Hat Enterprise Linux CoreOS during
...
+the OpenShift Ingress Controller should also accept for HTTP and HTTPS traffic.
+An administrator typically configures their Ingress Controller to be visible
+outside the cluster firewall, and may also add additional security, caching, or
+traffic controls on the Service content.
+
+=== Ingress Controller
+
+Usage: Ingress Controller(s)
+
+A resource that forwards traffic to endpoints of Services. Ingress Controller
```
Comment (mburke5678) [inline contributing_to_docs/term_glossary.adoc]: To be *really* picky, s/Ingress Controller/An Ingress Controller

**Comment by mburke5678** [comment]
@bmcelvee A tiny nit that you can ignore, being this an internal reference doc. LGTM

**Comment by bmcelvee** [comment]
Thank you, @mburke5678!

**Pull request merged**